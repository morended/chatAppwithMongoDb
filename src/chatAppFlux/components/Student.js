"use strict";

import React from 'react';
import ReactDOM from 'react-dom';
import chatActions from '../action/chatActions'

class Student extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      message: ""
    };
  }

  _handleOnChange = (e) => {
     this.setState({
       message:e.target.value
     });
   }

   _handleClick = (e) => {
      e.preventDefault();
      chatActions.updateMessages({
        id: this.props.id,
        message: this.state.message
      });

      this.setState({
        message: ""
      });
   }

   render() {
     return (
        <div>
          Student {this.props.id}:
          <input
            type="text"
            value={this.state.message}
            onChange={this._handleOnChange}/>

          <button
            onClick={this._handleClick}>
            Send
          </button>
        </div>
     );
  }
}

export default Student;
