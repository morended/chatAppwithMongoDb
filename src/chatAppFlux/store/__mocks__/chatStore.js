const chatStoreMock = jest.genMockFromModule('./../chatStore.js');

module.exports = chatStoreMock;
