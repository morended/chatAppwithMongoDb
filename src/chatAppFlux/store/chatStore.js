import alt from "../alt.js"

import chatActions from '../action/chatActions.js'

class chatStore {
  constructor() {
    this.messages = [];

    this.bindListeners({
      handleMessage: chatActions.UPDATE_MESSAGES
    });
  }

  handleMessage(messages) {
    console.log("hiii");
    console.log(messages.data);
    
    this.messages = messages.data;


    
  }
}

export default alt.createStore(chatStore);