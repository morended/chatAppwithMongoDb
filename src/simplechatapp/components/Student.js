"use strict";

import React from 'react';
import ReactDOM from 'react-dom';

class Student extends React.Component {
  constructor() {
    super()
    this.state = {
      message: ""
    }
    this._handleClick=this._handleClick.bind(this);
  this._handleOnChange=this._handleOnChange.bind(this);
  }
  _handleOnChange(event){

   this.setState({message:event.target.value})
 }
 _handleClick(){
   this.props.messages(this.props.id+this.state.message);
   this.state.message="";
 }
  render() {


return(
  <div>
{this.props.id}<input type="text"  value={this.state.message} onChange={this._handleOnChange}/>
</div>
<input type="Button" value="send"  onClick={this._handleClick}/>
  </div>
  )}

export default Student;