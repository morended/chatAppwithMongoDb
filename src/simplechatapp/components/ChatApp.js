"use strict";

import React from 'react';
import ReactDOM from 'react-dom';
import Student from './Student.js';

class ChatApp extends React.Component {
  constructor() {
    super();
    this.state = {
      messages: []
    }
      this.sendMessage=this.sendMessage.bind(this);

  }
  sendMessage(message){
    let smessages = this.state.messages;
      smessages.push(message);
      this.setState({
        messages:smessages
      })
  }
  render() {
    return (
      <div>
      <Student id="Student1 :" messages={this.sendMessage}/>
      <Student id="Student2 :" messages={this.sendMessage}/><br/>
      {this.state.messages.map((message, index) => (
       <p key={index}>{message}</p>
     ))}

      </div>

    );
  }
}
export default ChatApp;
