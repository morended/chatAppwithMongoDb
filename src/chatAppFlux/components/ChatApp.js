"use strict";

import React from 'react';
import ReactDOM from 'react-dom';
import Student from './Student.js';
import chatStore from '../store/chatStore.js'

class ChatApp extends React.Component {
  constructor() {
    super();
    this.state = {
      messages: []
    };
  }

  componentDidMount() {
    chatStore.listen(this.onChange)
  }

  componentWillUnmount() {
    chatStore.unlisten(this.onChange)
  }

  onChange = (state) => {
    this.setState({
      messages:state.messages
    });
  }

  render() {
    return (
      <div>
        <br/>
        <Student id="1"/>
        <Student id="2"/><br/>

        <h3>Chat History: </h3>

         {this.state.messages.map((msg, index) => (
           <p key={index}><b>Student {msg.id}</b>: {msg.message} </p>
         ))}
      </div>

    );
  }
}
export default ChatApp;
