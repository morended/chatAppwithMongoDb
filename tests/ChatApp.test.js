import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';

import ChatApp from '../src/chatAppFlux/components/ChatApp.js';
import Student from '../src/chatAppFlux/components/Student.js';
import chatStore from '../src/chatAppFlux/store/chatStore.js';

//mock chatStore
jest.mock('../src/chatAppFlux/store/chatStore.js');

describe('ChatApp component (snapshot)', () => {

  it('renders correctly', () => {
    const component = renderer.create(
      <ChatApp/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('ChatApp component', () => {
  let component;
  beforeEach(() => {
    component = shallow(
        <ChatApp/>
    );
  });

  it('renders two Student components', () => {
    expect(component.find('Student').length).toEqual(2);
  });

  it('has the chat history header', () => {
    expect(component.find('h3').length).toEqual(1);
  });

  it('initially chat history is empty', () => {
    expect(component.find('p').length).toEqual(0);
  });

  it('shows correct number of messages in history', () => {
    let msgs = [
      { id: "1", message: "a"},
      { id: "2", message: "b"},
      { id: "1", message: "c"}
    ];
    component.setState({ messages: msgs});
    expect(component.find('p').length).toEqual(msgs.length);
    const html = component.html();
    expect(html).toContain(msgs[0].message);
    expect(html).toContain(msgs[1].message);
    expect(html).toContain(msgs[2].message);
  });

  it('updates state on change', () => {
    let testMsgs = [
      { id: "1", message: "a"}
    ];
    component.instance().onChange({ messages: testMsgs });
    expect(component.find('p').length).toEqual(testMsgs.length);
  });

  it('listens to chatstore on mount', () => {
    component = mount(<ChatApp/>);
    expect(chatStore.listen).toHaveBeenCalled();
  });

  it('unlistens to chatstore on unmount', () => {
    component = mount(<ChatApp/>);
    component.unmount();
    expect(chatStore.unlisten).toHaveBeenCalled();
  });

  describe('lifecyle', () => {
    beforeEach(() => {
      ChatApp.prototype.componentDidMount = jest.fn();
      ChatApp.prototype.componentWillUnmount = jest.fn();
      component = mount(<ChatApp/>);
    });

    it('calls componentDidMount once when mounted', () => {
      expect(ChatApp.prototype.componentDidMount).toHaveBeenCalledTimes(1);
    });

    it('calls componentWillUnmount once when unmounted', () => {
      component.unmount();
      expect(ChatApp.prototype.componentWillUnmount).toHaveBeenCalledTimes(1);
    });
  });
});
