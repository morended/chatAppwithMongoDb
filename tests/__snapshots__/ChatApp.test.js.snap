// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatApp component (snapshot) renders correctly 1`] = `
<div>
  <br />
  <div>
    Student 
    1
    :
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Send
    </button>
  </div>
  <div>
    Student 
    2
    :
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Send
    </button>
  </div>
  <br />
  <h3>
    Chat History: 
  </h3>
</div>
`;
